### DISCUSSION TITLE
can't we delete all the mailboxes at once?

### DISCUSSION THREAD

**Discussion by kiranparajuli589** [inbucket/inbucket#300] [Q&A]
By using the REST API we can only purge emails from a specific mailbox at once. Is there any way to cleanup the whole mailboxes at once using some REST API?

This would be very useful to do cleanups in before|after scenarios for the e2e tests.

**Comment by jhillyerd** [comment]
There is not currently an API to do that.  It could be implemented, but is a bit tricky since it would require inbucket to scan the disk looking for mailboxes (it doesn't maintain an in memory list when using file storage mode).  Feel free to submit a feature request issue.

If you are looking for a quick solution, you can just rm -rf the `mail` subdirectory of the file storage directory.

**Reply by kiranparajuli589** [reply]
Thank you I've tried with your trick. Navigated to docker exec arena and checked the `/storage` dir there were some dirs-sub dirs and at last message info. Instantly did:
```sh
cd /storage
rm -rf *
```
Still I have the old messages :(. Also If I restart the docker container,,,,the mailbox seems to be reset.  I'll try to make a PR for this :) Do I need to create a ticket to add a PR?

**Reply by jhillyerd** [reply]
No issue needed for a PR.
